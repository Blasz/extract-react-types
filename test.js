// @flow
'use strict';

const extractReactTypes = require('./');
const stripIndent = require('strip-indent');

const TESTS = [{
  name: 'boolean',
  typeSystem: 'flow',
  code: `
    class Component extends React.Component<{ foo: boolean }> {
      // ...
    }
  `
}, {
  name: 'string',
  typeSystem: 'flow',
  code: `
    class Component extends React.Component<{ foo: string }> {
      // ...
    }
  `
}];

for (let testCase of TESTS) {
  test(testCase.name, () => {
    let code = stripIndent(testCase.code);
    let result = extractReactTypes(code, testCase.typeSystem);
    expect(result).toMatchSnapshot();
  });
}
