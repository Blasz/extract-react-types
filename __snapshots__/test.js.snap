// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`boolean 1`] = `
Object {
  "classes": Array [
    Object {
      "kind": "object",
      "props": Array [
        Object {
          "key": "foo",
          "kind": "property",
          "value": Object {
            "kind": "boolean",
          },
        },
      ],
    },
  ],
  "kind": "program",
}
`;

exports[`string 1`] = `
Object {
  "classes": Array [
    Object {
      "kind": "object",
      "props": Array [
        Object {
          "key": "foo",
          "kind": "property",
          "value": Object {
            "kind": "string",
          },
        },
      ],
    },
  ],
  "kind": "program",
}
`;
